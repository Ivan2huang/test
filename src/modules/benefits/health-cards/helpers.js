/* eslint-disable import/prefer-default-export */
export const getQRCodeXML = ({
  fullName,
  membershipNumber,
  certificateNumber,
  insurerCode,
  insurerName,
  insurerType,
  policyNumber,
  expiryDate,
}) =>
  [
    '<?xml version="1.0"?>',
    '<medical_card>',
    '<Card_type>',
    insurerType,
    '</Card_type>',
    '<insurance_company_code>',
    insurerCode,
    '</insurance_company_code>',
    '<insurance_company_name>',
    insurerName,
    '</insurance_company_name>',
    '<insured_name>',
    fullName,
    '</insured_name>',
    '<policy_no>',
    policyNumber,
    '</policy_no>',
    '<certificate_no>',
    certificateNumber,
    '</certificate_no>',
    '<membership_no>',
    membershipNumber,
    '</membership_no>',
    '<expiry_date>',
    expiryDate,
    '</expiry_date>',
    '</medical_card>',
  ].join('');
