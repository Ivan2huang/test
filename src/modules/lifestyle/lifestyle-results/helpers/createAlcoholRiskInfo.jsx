import React from 'react';

import { AlcoholIcon, NoAlcoholIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createAlcoholRiskInfo = (intl, theme, alcoholRisk) => {
  const alcoholRiskMapping = {
    None: `${theme.lifeStyleResultOK}`,
    RegularDrinker: `${theme.lifeStyleResultWarning}`,
    HeavyDrinker: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'alcohol',
    color: alcoholRiskMapping[alcoholRisk],
    CardIcon:
      alcoholRisk === 'None' ? (
        <NoAlcoholIcon />
      ) : (
        <AlcoholIcon color={alcoholRiskMapping[alcoholRisk]} />
      ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.alcoholRiskCard.title',
      'Alcohol',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.alcoholRiskCard.subTitle.${alcoholRisk}`,
      alcoholRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.alcoholRiskCard.description.${alcoholRisk}`,
      alcoholRisk,
    ),
  };
};

export default createAlcoholRiskInfo;
