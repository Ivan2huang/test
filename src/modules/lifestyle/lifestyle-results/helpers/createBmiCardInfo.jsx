import React from 'react';

import Typography from '../../../../uiComponents/Typography';
import { formatMessage } from '../../../../helpers/helpers';

export const formatBmiScore = bmiScore => {
  const score = parseFloat(bmiScore);
  if (Number.isNaN(score)) {
    return '';
  }
  return (+bmiScore).toFixed(1);
};

const createBmiCardInfo = (intl, theme, bmiScore, bmiClass) => {
  const bmiColorMapping = {
    Healthy: `${theme.lifeStyleResultOK}`,
    Overweight: `${theme.lifeStyleResultWarning}`,
    Obese: `${theme.lifeStyleResultAlarm}`,
    Underweight: `${theme.lifeStyleResultWarning}`,
  };

  return {
    id: 'bmi',
    color: bmiColorMapping[bmiClass],
    CardIcon: (
      <Typography type="style-1" color={theme.grey3} fontWeight="bold">
        {formatBmiScore(bmiScore)}
      </Typography>
    ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.bmiCard.title',
      'BMI',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.bmiCard.subTitle.${bmiClass}`,
      bmiClass,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.bmiCard.description.${bmiClass}`,
      bmiClass,
    ),
  };
};

export default createBmiCardInfo;
