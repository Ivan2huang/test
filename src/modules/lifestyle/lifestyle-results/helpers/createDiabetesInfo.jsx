import React from 'react';

import { HeartIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createDiabetesInfo = (intl, theme, diabetesRisk) => {
  const alcoholRiskMapping = {
    Low: `${theme.lifeStyleResultOK}`,
    High: `${theme.lifeStyleResultWarning}`,
    VeryHigh: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'diabetes',
    color: alcoholRiskMapping[diabetesRisk],
    CardIcon: <HeartIcon color={alcoholRiskMapping[diabetesRisk]} />,
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.diabetesCard.title',
      'Diabetes',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.diabetesCard.subTitle.${diabetesRisk}`,
      diabetesRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.diabetesCard.description.${diabetesRisk}`,
      diabetesRisk,
    ),
  };
};

export default createDiabetesInfo;
