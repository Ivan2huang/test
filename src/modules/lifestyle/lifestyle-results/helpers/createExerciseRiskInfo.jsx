import React from 'react';

import { ExerciseIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createExerciseRiskInfo = (intl, theme, exerciseRisk) => {
  const exerciseRiskMapping = {
    High: `${theme.lifeStyleResultOK}`,
    LowToModerate: `${theme.lifeStyleResultWarning}`,
    VeryLow: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'exercise',
    color: exerciseRiskMapping[exerciseRisk],
    CardIcon: <ExerciseIcon color={exerciseRiskMapping[exerciseRisk]} />,
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.exerciseRiskCard.title',
      'Exercise',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.exerciseRiskCard.subTitle.${exerciseRisk}`,
      exerciseRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.exerciseRiskCard.description.${exerciseRisk}`,
      exerciseRisk,
    ),
  };
};

export default createExerciseRiskInfo;
