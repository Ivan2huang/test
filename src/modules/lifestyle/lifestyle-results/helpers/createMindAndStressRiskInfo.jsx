import React from 'react';

import { StressIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createMindAndStressRiskInfo = (intl, theme, mindAndStressRisk) => {
  const mindAndStressRiskMapping = {
    Normal: `${theme.lifeStyleResultOK}`,
    BeCareful: `${theme.lifeStyleResultWarning}`,
    AtRisk: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'mindAndStress',
    color: mindAndStressRiskMapping[mindAndStressRisk],
    CardIcon: (
      <StressIcon color={mindAndStressRiskMapping[mindAndStressRisk]} />
    ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.mindAndStressRiskCard.title',
      'Mind & stress',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.mindAndStressRiskCard.subTitle.${mindAndStressRisk}`,
      mindAndStressRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.mindAndStressRiskCard.description.${mindAndStressRisk}`,
      mindAndStressRisk,
    ),
  };
};

export default createMindAndStressRiskInfo;
