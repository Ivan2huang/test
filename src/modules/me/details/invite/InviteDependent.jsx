import React, { useEffect } from 'react';
import * as PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import { compose } from 'redux';

import { Field, reduxForm, SubmissionError } from 'redux-form';
import { Box, Card, CardContent, withStyles, Hidden } from '@material-ui/core';
import TrackingButton from '../../../../uiComponents/TrackingButton';
import Grid from '../../../../uiComponents/Grid';
import GridItem from '../../../../uiComponents/GridItem';
import Typography from '../../../../uiComponents/Typography';
import ButtonGroup from '../../../../uiComponents/ButtonGroup';
import NavigationBackButton from '../../../../uiComponents/NavigationBackButton';
import renderTextField from '../../../../utils/renderTextField';
import { formatMessage, navigateTo } from '../../../../helpers/helpers';
import Images from '../../../../constants/images';
import {
  validateEmail,
  validateRequired,
  validateEmailsMatch,
  validateDateOfBirth,
} from './validation';
import renderDatePicker from '../../../../utils/renderDatePicker';
import paths from '../../../../helpers/paths';
import { DETAILS_ID } from '../../constant';
import { CATEGORIES } from '../../../../constants/analytics';

const Styles = () => ({
  image: {
    width: '100%',
  },
});

const InviteDependent = ({
  intl,
  handleSubmit,
  change,
  dependentName,
  emailAlreadyTaken,
  hasDefaultDateOfBirth,
  submissionErrors,
  setFormErrorIfEamilAlreadyTaken,
}) => {
  useEffect(() => {
    if (emailAlreadyTaken && emailAlreadyTaken.errorState) {
      setFormErrorIfEamilAlreadyTaken({
        formName: 'inviteDependent',
        fieldName: 'dependentEmail',
        errorMessage: formatMessage(
          intl,
          'me.tabs.myDetails.invite.dependent.validations.accountAlreadyInUse',
          'Email already exist. Try a new email address',
        ),
      });
    }
  }, [emailAlreadyTaken]);

  const onDateOfBirthChange = date => {
    change('inviteDependent.dateOfBirthInput', date);
  };

  const DateOfBirthMaximum = () => {
    return new Date();
  };

  const DateOfBirthMinimum = () => {
    const date = new Date();
    date.setFullYear(date.getFullYear() - 110);
    return date;
  };

  const navigateToMyDetails = () => navigateTo(paths.common[DETAILS_ID]);

  return (
    <Box
      mt={{
        xs: 3,
        sm: 3,
      }}
    >
      <NavigationBackButton
        testId="btn-back-mydetails-inviteDependent"
        onClick={navigateToMyDetails}
      >
        {formatMessage(
          intl,
          'me.tabs.myDetails.invite.dependent.button.backToMyDetails',
          '< My details',
        )}
      </NavigationBackButton>
      <Card>
        <CardContent>
          <Box mt={{ xs: 0, md: 10 }} mb={10}>
            <Grid>
              <GridItem offset={{ md: 1 }} columns={{ md: 3 }}>
                <Box mb={4} mt={{ md: 5 }}>
                  <img src={Images.INVITE_DEPENDENT} alt="background" />
                </Box>
              </GridItem>
              <GridItem
                offset={{ sm: 1, md: 1 }}
                columns={{ xs: 12, sm: 10, md: 5 }}
              >
                <Hidden implementation="css" smUp>
                  <Box mb={4}>
                    <Typography type="style-2">
                      {formatMessage(
                        intl,
                        'me.tabs.myDetails.invite.dependent.header.title.sm',
                        'My details',
                      )}
                    </Typography>
                  </Box>
                </Hidden>
                <Hidden implementation="css" smDown>
                  <Box mb={4}>
                    <Typography type="style-1">
                      {formatMessage(
                        intl,
                        'me.tabs.myDetails.invite.dependent.header.title',
                        'Invite',
                        { dependentName },
                      )}
                    </Typography>
                  </Box>
                </Hidden>
                <Box mb={8}>
                  <Typography type="style-6">
                    {formatMessage(
                      intl,
                      'me.tabs.myDetails.invite.dependent.instructionsFirst',
                      'Invite dependents to HSBC Benefits so they can access the dependent app and website.',
                    )}
                  </Typography>
                </Box>
                <Box mb={8}>
                  <Typography type="style-6">
                    {formatMessage(
                      intl,

                      'me.tabs.myDetails.invite.dependent.instructionsSecond',
                      "Provide dependent's email address and he/she will receive instructions to create a dependent account.",
                      { dependentName },
                    )}
                  </Typography>
                </Box>
                <Box mb={8}>
                  <form onSubmit={handleSubmit} autoComplete="off">
                    <Box mt={4} mb={6}>
                      <Field
                        name="dateOfBirth"
                        label={formatMessage(
                          intl,
                          'me.tabs.myDetails.invite.dependent.label.dateOfBirth',
                          'Date of birth',
                        )}
                        errorMessage={
                          submissionErrors && submissionErrors.dateOfBirth
                            ? submissionErrors.dateOfBirth
                            : formatMessage(
                                intl,
                                'dateofBirth.validations.required',
                                'Enter date of birth',
                              )
                        }
                        testId="dependent-invite-dateofbirth"
                        minDate={DateOfBirthMinimum()}
                        maxDate={DateOfBirthMaximum()}
                        onChange={onDateOfBirthChange}
                        disabled={hasDefaultDateOfBirth}
                        component={renderDatePicker}
                        validate={[validateRequired]}
                      />
                      <Field
                        name="dependentEmail"
                        label={formatMessage(
                          intl,
                          'me.tabs.myDetails.invite.dependent.email',
                          'Dependent email address',
                        )}
                        testId="dependent-invite-email"
                        component={renderTextField}
                        validate={[validateRequired, validateEmail]}
                      />
                      <Field
                        name="confirmEmail"
                        label={formatMessage(
                          intl,
                          'me.tabs.myDetails.invite.dependent.confirm.email',
                          'Confirm email address',
                        )}
                        testId="dependent-confirm-invite-email"
                        component={renderTextField}
                        validate={[validateRequired, validateEmail]}
                      />
                    </Box>
                    <ButtonGroup>
                      <TrackingButton
                        type="submit"
                        variant="contained"
                        color="primary"
                        data-testid="btn-submit-inviteDependent"
                        trackingData={{
                          category: CATEGORIES.DETAILS_PAGE,
                          action: 'Invite dependents',
                        }}
                      >
                        {formatMessage(
                          intl,
                          'me.tabs.myDetails.invite.dependent.button.submit',
                          'Send Invite',
                        )}
                      </TrackingButton>
                    </ButtonGroup>
                  </form>
                </Box>
              </GridItem>
            </Grid>
          </Box>
        </CardContent>
      </Card>
    </Box>
  );
};

const onSubmit = (values, _, props) => {
  const { intl, relationshipCategory } = props;
  const { dateOfBirth, dependentEmail, confirmEmail } = values;

  const errors = {
    confirmEmail: validateEmailsMatch(intl, dependentEmail, confirmEmail),
    dateOfBirth: validateDateOfBirth(intl, dateOfBirth, relationshipCategory),
  };

  if (errors.confirmEmail !== '' || errors.dateOfBirth !== '') {
    throw new SubmissionError(errors);
  }
  const loaderMessage = props.intl.formatHTMLMessage({
    id: 'me.tabs.myDetails.invite.dependent.submitLoaderMessage',
    defaultMessage: 'Sending an invite...<br/>Do not close or refresh',
  });

  const dependentData = {
    dependentId: props.memberId,
    dependentName: props.dependentName,
    hasDefaultDateOfBirth: props.hasDefaultDateOfBirth,
    dependentEmail,
    dateOfBirth,
  };
  props.inviteDependentAction(dependentData, loaderMessage);
};

InviteDependent.propTypes = {
  intl: PropTypes.shape({}).isRequired,
  handleSubmit: PropTypes.func.isRequired,
  change: PropTypes.func.isRequired,
  dependentName: PropTypes.string.isRequired,
  emailAlreadyTaken: PropTypes.shape({}),
  hasDefaultDateOfBirth: PropTypes.bool.isRequired,
  submissionErrors: PropTypes.shape({}),
  setFormErrorIfEamilAlreadyTaken: PropTypes.func.isRequired,
};

InviteDependent.defaultProps = {
  emailAlreadyTaken: {},
  submissionErrors: {},
};

export default compose(
  injectIntl,
  withStyles(Styles),
  reduxForm({
    enableReinitialize: true,
    destroyOnUnmount: false,
    keepDirtyOnReinitialize: false,
    onSubmit,
    form: 'inviteDependent',
  }),
)(InviteDependent);
