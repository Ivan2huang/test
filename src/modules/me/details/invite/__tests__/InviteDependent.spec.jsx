/* eslint-disable react/prop-types */

import React from 'react';
import { render, fireEvent, wait, getByText } from '@testing-library/react';
import withIntl from '../../../../../i18n/withIntlProvider';
import withTheme from '../../../../../themes/withThemeProvider';
import withRedux from '../../../../../redux/withReduxProvider';
import InviteDependent from '../InviteDependent';
import {
  validateEmail,
  validateRequired,
  validateEmailsMatch,
  validateDateOfBirth,
} from '../validation';
import { navigateTo } from '../../../../../helpers/helpers';

jest.mock('../../../../../helpers/helpers', () => ({
  formatMessage: jest.fn((intl, id, defaultMessage) => defaultMessage),
  navigateTo: jest.fn(),
}));

jest.mock('../validation');
validateRequired.mockImplementation(() => '');
validateEmail.mockImplementation(() => '');
validateEmailsMatch.mockImplementation(() => '');
validateDateOfBirth.mockImplementation(() => '');

jest.mock('../../../../../uiComponents/Typography', () => props => (
  <div {...props}>Typography Component</div>
));

jest.mock(
  '../../../../../uiComponents/GridItem',
  () => ({ columns, children }) => (
    <div data-xs={columns.xs} data-sm={columns.sm} data-md={columns.md}>
      <span>GridItem Component</span>
      <div>{children}</div>
    </div>
  ),
);

jest.mock('../../../../../uiComponents/Grid', () => ({ children }) => (
  <div>
    <span>Grid Component</span>
    <div>{children}</div>
  </div>
));

jest.mock(
  '../../../../../uiComponents/ButtonGroup',
  () => ({ children, inverse, ...rest }) => (
    <div data-inverse={inverse} {...rest}>
      <span>Button Group Component</span>
      {children}
    </div>
  ),
);

const fireUpdateDate = (dateOfBirthButton, value) => {
  fireEvent.click(dateOfBirthButton);
  const dialog = document.querySelector('[role=dialog]');
  const selectDate = getByText(dialog, value);
  fireEvent.click(selectDate);

  const okButton = getByText(dialog, 'OK');
  fireEvent.click(okButton);
};

describe('InviteDependentComponent', () => {
  it('should match the snapshot', () => {
    const componentProps = {
      dependent: {
        fullName: 'Helen Tan',
        memberId: '111',
      },
      dependentName: 'Test',
      hasDefaultDateOfBirth: false,
      setFormErrorIfEamilAlreadyTaken: jest.fn(),
    };
    const Component = withRedux(withIntl(InviteDependent));
    const result = render(<Component {...componentProps} />);

    expect(result.container).toMatchSnapshot();
  });

  it('should call onSubmit on form submit', async () => {
    const inviteData = {
      dependentId: '111',
      dependentName: 'Helen Tan',
      dependentEmail: 'test@test.com',
    };

    const componentProps = {
      dependentName: 'Helen Tan',
      memberId: '111',
      inviteDependentAction: jest.fn(),
      hasDefaultDateOfBirth: false,
      setFormErrorIfEamilAlreadyTaken: jest.fn(),
    };
    const Component = withRedux(withIntl(withTheme(InviteDependent)));
    const result = render(<Component {...componentProps} />);

    const dateOfBirthButton = result
      .getByTestId('dependent-invite-dateofbirth')
      .querySelector('button');
    const emailInput = result
      .getByTestId('dependent-invite-email')
      .querySelector('input');
    const confirmEmailInput = result
      .getByTestId('dependent-confirm-invite-email')
      .querySelector('input');
    const button = result.getByTestId('btn-submit-inviteDependent');

    fireUpdateDate(dateOfBirthButton, '10');
    fireEvent.change(emailInput, { target: { value: 'test@test.com' } });
    fireEvent.change(confirmEmailInput, { target: { value: 'test@test.com' } });

    fireEvent.click(button, {});

    await wait(() => {
      expect(componentProps.inviteDependentAction).toHaveBeenCalledTimes(1);
      expect(componentProps.inviteDependentAction).toHaveBeenCalledWith(
        expect.objectContaining(inviteData),
        'Sending an invite...<br/>Do not close or refresh',
      );
    });
  });

  it('should show error if email and confirm email do not match', async () => {
    validateEmailsMatch.mockImplementation(() => 'Emails do not match');
    const componentProps = {
      dependent: {
        fullName: 'Helen Tan',
        memberId: '111',
      },
      inviteDependentAction: jest.fn(),
      hasDefaultDateOfBirth: false,
      setFormErrorIfEamilAlreadyTaken: jest.fn(),
      dependentName: 'test',
    };
    const Component = withRedux(withIntl(withTheme(InviteDependent)));
    const result = render(<Component {...componentProps} />);

    const dateOfBirthButton = result
      .getByTestId('dependent-invite-dateofbirth')
      .querySelector('button');
    const emailInput = result
      .getByTestId('dependent-invite-email')
      .querySelector('input');
    const confirmEmailInput = result
      .getByTestId('dependent-confirm-invite-email')
      .querySelector('input');
    const button = result.getByTestId('btn-submit-inviteDependent');

    fireUpdateDate(dateOfBirthButton, '10');
    fireEvent.change(emailInput, { target: { value: 'test@test.com' } });
    fireEvent.change(confirmEmailInput, {
      target: { value: 'test2@test.com' },
    });
    fireEvent.click(button, {});

    await wait(() => {
      expect(result.getByText('Emails do not match')).toBeDefined();
    });
  });

  it('should call set form error when email already taken props has errorState', async () => {
    const componentProps = {
      dependent: {
        fullName: 'Helen Tan',
        memberId: '111',
      },
      inviteDependentAction: jest.fn(),
      emailAlreadyTaken: {
        errorState: true,
      },
      setFormErrorIfEamilAlreadyTaken: jest.fn(),
      hasDefaultDateOfBirth: false,
      dependentName: 'test',
    };

    const Component = withRedux(withIntl(withTheme(InviteDependent)));
    render(<Component {...componentProps} />);

    expect(
      componentProps.setFormErrorIfEamilAlreadyTaken,
    ).toHaveBeenCalledTimes(1);
    expect(componentProps.setFormErrorIfEamilAlreadyTaken).toHaveBeenCalledWith(
      {
        formName: 'inviteDependent',
        fieldName: 'dependentEmail',
        errorMessage: 'Email already exist. Try a new email address',
      },
    );
  });

  it('should not call set form error when email already taken props has no errorState', async () => {
    const componentProps = {
      dependent: {
        fullName: 'Helen Tan',
        memberId: '111',
      },
      inviteDependentAction: jest.fn(),
      emailAlreadyTaken: {
        errorState: false,
      },
      setFormErrorIfEamilAlreadyTaken: jest.fn(),
      hasDefaultDateOfBirth: false,
      dependentName: 'test',
    };

    const Component = withRedux(withIntl(withTheme(InviteDependent)));
    render(<Component {...componentProps} />);

    expect(
      componentProps.setFormErrorIfEamilAlreadyTaken,
    ).toHaveBeenCalledTimes(0);
  });

  it('should show error if date of birth is not entered', async () => {
    validateRequired.mockImplementation(() => 'Required');
    const componentProps = {
      dependent: {
        fullName: 'Helen Tan',
        memberId: '111',
        relationship: 'child',
      },
      dependentName: 'test',
      hasDefaultDateOfBirth: false,
      inviteDependentAction: jest.fn(),
      setFormErrorIfEamilAlreadyTaken: jest.fn(),
    };
    const Component = withRedux(withIntl(withTheme(InviteDependent)));
    const result = render(<Component {...componentProps} />);

    const emailInput = result
      .getByTestId('dependent-invite-email')
      .querySelector('input');
    const confirmEmailInput = result
      .getByTestId('dependent-confirm-invite-email')
      .querySelector('input');
    const button = result.getByTestId('btn-submit-inviteDependent');

    fireEvent.change(emailInput, { target: { value: 'test@test.com' } });
    fireEvent.change(confirmEmailInput, { target: { value: 'test@test.com' } });
    fireEvent.click(button, {});

    await wait(() => {
      expect(result.getByText('Enter date of birth')).toBeDefined();
    });
  });

  it('should show error if date of birth has submission error', async () => {
    validateRequired.mockImplementation(() => 'Required');
    const componentProps = {
      hasDefaultDateOfBirth: false,
      submissionErrors: {
        dateOfBirth: 'Date of birth error',
      },
      inviteDependentAction: jest.fn(),
      setFormErrorIfEamilAlreadyTaken: jest.fn(),
      dependentName: 'test',
    };
    const Component = withRedux(withIntl(withTheme(InviteDependent)));
    const result = render(<Component {...componentProps} />);
    const button = result.getByTestId('btn-submit-inviteDependent');
    fireEvent.click(button, {});

    await wait(() => {
      expect(result.getByText('Date of birth error')).toBeDefined();
    });
  });

  it('should navigate to me page when back to my details button is clicked', async () => {
    const componentProps = {
      dependent: {
        fullName: 'Helen Tan',
        memberId: '111',
      },
      dependentName: 'test',
      inviteDependentAction: jest.fn(),
      hasDefaultDateOfBirth: false,
      setFormErrorIfEamilAlreadyTaken: jest.fn(),
    };
    const Component = withRedux(withIntl(InviteDependent));
    const result = render(<Component {...componentProps} />);

    const button = result.getByTestId('btn-back-mydetails-inviteDependent');
    fireEvent.click(button, {});

    await wait(() => {
      expect(navigateTo).toHaveBeenCalledTimes(1);
      expect(navigateTo).toHaveBeenCalledWith('/me/details');
    });
  });
});
