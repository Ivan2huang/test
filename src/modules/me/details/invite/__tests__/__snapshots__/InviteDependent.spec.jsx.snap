// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteDependentComponent should match the snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <div
      class="MuiBox-root MuiBox-root-6"
    >
      <button
        class="makeStyles-backBtn-4 makeStyles-backBtn-5"
        data-testid="btn-back-mydetails-inviteDependent"
        type="button"
      >
        &lt; My details
      </button>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <div
          class="MuiBox-root MuiBox-root-36"
        >
          <div>
            <span>
              Grid Component
            </span>
            <div>
              <div
                data-md="3"
              >
                <span>
                  GridItem Component
                </span>
                <div>
                  <div
                    class="MuiBox-root MuiBox-root-37"
                  >
                    <img
                      alt="background"
                      src="/static/invite_dependent.svg"
                    />
                  </div>
                </div>
              </div>
              <div
                data-md="5"
                data-sm="10"
                data-xs="12"
              >
                <span>
                  GridItem Component
                </span>
                <div>
                  <div
                    class="PrivateHiddenCss-smUp-42"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-53"
                    >
                      <div
                        type="style-2"
                      >
                        Typography Component
                      </div>
                    </div>
                  </div>
                  <div
                    class="PrivateHiddenCss-smDown-43"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-54"
                    >
                      <div
                        type="style-1"
                      >
                        Typography Component
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root MuiBox-root-55"
                  >
                    <div
                      type="style-6"
                    >
                      Typography Component
                    </div>
                  </div>
                  <div
                    class="MuiBox-root MuiBox-root-56"
                  >
                    <div
                      type="style-6"
                    >
                      Typography Component
                    </div>
                  </div>
                  <div
                    class="MuiBox-root MuiBox-root-57"
                  >
                    <form
                      autocomplete="off"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-58"
                      >
                        <span>
                          <div
                            class="MuiFormControl-root MuiFormControl-fullWidth"
                            data-testid="dependent-invite-dateofbirth"
                          >
                            <label
                              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                              data-shrink="false"
                              for="input-dateOfBirth"
                            >
                              Date of birth
                            </label>
                            <div
                              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                              label="Date of birth"
                            >
                              <input
                                aria-describedby="helptext-dateOfBirth"
                                aria-invalid="false"
                                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                                disabled="disabled"
                                id="input-dateOfBirth"
                                name="dateOfBirth"
                                type="text"
                                value=""
                              />
                              <div
                                class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                              >
                                <button
                                  aria-describedby="helptext-dateOfBirth"
                                  class="MuiButtonBase-root MuiIconButton-root"
                                  tabindex="0"
                                  type="button"
                                >
                                  <span
                                    class="MuiIconButton-label"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      class="MuiSvgIcon-root"
                                      focusable="false"
                                      role="presentation"
                                      style="width: 1.125rem; height: 1.125rem;"
                                      viewBox="0 0 24 24"
                                    >
                                      <path
                                        d="M20 1.33333V0H18.4V1.33333H5.6V0H4V1.33333H0V24H18.9333L24 18.9333V1.33333H20ZM18.9333 21.7373V18.9333H21.7373L18.9333 21.7373ZM22.4 17.3333H17.3333V22.4H1.6V8H22.4V17.3333ZM22.4 6.4H1.6V2.93333H4V4.66667H5.6V2.93333H18.4V4.66667H20V2.93333H22.4V6.4ZM11.7333 13.372V20H13.3333V10.628L9.33733 12.8893L10.1253 14.2827L11.7333 13.372Z"
                                        fill="#333333"
                                      />
                                    </svg>
                                  </span>
                                  <span
                                    class="MuiTouchRipple-root"
                                  />
                                </button>
                              </div>
                              <fieldset
                                aria-hidden="true"
                                class="PrivateNotchedOutline-root-145 MuiOutlinedInput-notchedOutline"
                                style="padding-left: 8px;"
                              >
                                <legend
                                  class="PrivateNotchedOutline-legend-146"
                                  style="width: 0.01px;"
                                >
                                  <span>
                                    ​
                                  </span>
                                </legend>
                              </fieldset>
                            </div>
                            <div
                              aria-live="polite"
                            />
                          </div>
                        </span>
                        <div
                          class="MuiFormControl-root MuiFormControl-fullWidth"
                          data-testid="dependent-invite-email"
                        >
                          <label
                            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                            data-shrink="false"
                            for="input-dependentEmail"
                          >
                            Dependent email address
                          </label>
                          <div
                            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                            label="Dependent email address"
                          >
                            <input
                              aria-describedby="helptext-dependentEmail"
                              aria-invalid="false"
                              class="MuiInputBase-input MuiOutlinedInput-input"
                              id="input-dependentEmail"
                              type="text"
                              value=""
                            />
                            <fieldset
                              aria-hidden="true"
                              class="PrivateNotchedOutline-root-145 MuiOutlinedInput-notchedOutline"
                              style="padding-left: 8px;"
                            >
                              <legend
                                class="PrivateNotchedOutline-legend-146"
                                style="width: 0.01px;"
                              >
                                <span>
                                  ​
                                </span>
                              </legend>
                            </fieldset>
                          </div>
                          <div
                            aria-live="polite"
                          />
                        </div>
                        <div
                          class="MuiFormControl-root MuiFormControl-fullWidth"
                          data-testid="dependent-confirm-invite-email"
                        >
                          <label
                            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                            data-shrink="false"
                            for="input-confirmEmail"
                          >
                            Confirm email address
                          </label>
                          <div
                            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                            label="Confirm email address"
                          >
                            <input
                              aria-describedby="helptext-confirmEmail"
                              aria-invalid="false"
                              class="MuiInputBase-input MuiOutlinedInput-input"
                              id="input-confirmEmail"
                              type="text"
                              value=""
                            />
                            <fieldset
                              aria-hidden="true"
                              class="PrivateNotchedOutline-root-145 MuiOutlinedInput-notchedOutline"
                              style="padding-left: 8px;"
                            >
                              <legend
                                class="PrivateNotchedOutline-legend-146"
                                style="width: 0.01px;"
                              >
                                <span>
                                  ​
                                </span>
                              </legend>
                            </fieldset>
                          </div>
                          <div
                            aria-live="polite"
                          />
                        </div>
                      </div>
                      <div>
                        <span>
                          Button Group Component
                        </span>
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                          data-testid="btn-submit-inviteDependent"
                          tabindex="0"
                          type="submit"
                        >
                          <span
                            class="MuiButton-label"
                          >
                            Send Invite
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
